diagnostic.md

```diff
@@ -36,6 +39,11 @@ Record your responses inside the fenced code blocks below each question.
 
     ```md
     <!-- your response here -->
+    first one should how the product and products are related, and 2nd just simply denote the path or url with the product, while the url is after products, it doesn't have to had any relationship with products.
+
+    in the 1st case, the product template can be loaded onto the {{outlet}} in products template, but not in the 2nd case.
+
+    If you loaded case 1, the products template will be loaded to the application, then product template can be loaded to the outlet section of products template.  case 2 will just load product template to the application by-passing products template.
```

Comment:
These two route definitions are identical  URL paths.  
Nested route definitions, do require a nested file structure, but one of the big distinctions is where Ember will render the content.  
The nested `'products'/'product'` instructs Ember to render
`app/products/product/template.hbs` in the `outlet` of the parent `app/products/template.hbs`.  
The non-nested `'product'` instructs Ember to render `app/product/template.hbs` in the context of the application template.